### PR TITLE
feat: add WiFi controller with AP support

### PR DESCRIPTION
**Pull request by Tasssadar**

### CODE REVIEW

File: main/platform/espWifi.cpp [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+            auto wifiNs = EspNvsKeyValue::open(KvNsWifiSsid);
+            if(!wifiNs) {
+                esp_wifi_clear_ap_list();
+                break;
+            }
+
+            wifi_ap_record_t rec;
+            const std::string not_exists_sentinel = "a";
+            while(true) {
+                auto err = esp_wifi_scan_get_ap_record(&rec);
```
Comment (cubicap) [inline main/platform/espWifi.cpp:342]: `esp_wifi_scan_get_ap_record` is not available in esp-idf 5.1

**Review by cubicap — DISMISSED** [review]
otherwise seems good

File: main/platform/espWifi.cpp (hunk 1 above)
Comment (Tasssadar) [reply main/platform/espWifi.cpp:342]: It is in 5.1.4

Comment (cubicap) [reply main/platform/espWifi.cpp:342]: it seemed like a quick change but updating idf seems more reasonable

**Pull request merged**